transform/operations_test.go

```diff
@@ -45,31 +46,55 @@ type opTests struct {
 func runOpTests(t *testing.T, opType func() transformOperation, tests []opTests) {
 
 	for _, test := range tests {
-		op := opType()
-		err := op.init(test.args)
+		runOpTest(t, opType, test)
+	}
+}
 
-		switch {
-		case test.wantInitErr && err != nil:
-			continue
-		case test.wantInitErr && err == nil:
-			t.Errorf("Test %q - got init error nil, want error", test.description)
-		case !test.wantErr && err != nil:
-			t.Errorf("Test %q - got init error, want nil: %v", test.description, err)
-		}
+func runOpTest(t *testing.T, opType func() transformOperation, test opTests) {
+	op, err := runOpTestInit(opType, test)
+	if test.wantInitErr && err != nil {
+		return
+	} else if err != nil {
+		t.Error(err)
+	}
+
+	_, err = runOpTestTransform(op, test)
+	if test.wantErr && err != nil {
+		return
+	} else if err != nil {
+		t.Error(err)
+	}
+}
 
-		got, err := op.transform(test.in)
+func runOpTestInit(opType func() transformOperation, test opTests) (transformOperation, error) {
+	op := opType()
+	err := op.init(test.args)
+	switch {
+	case test.wantInitErr && err != nil:
+		return op, err
+	case test.wantInitErr && err == nil:
+		return op, fmt.Errorf("Test %q - got init error nil, want error", test.description)
+	case !test.wantErr && err != nil:
+		return op, fmt.Errorf("Test %q - got init error, want nil: %v", test.description, err)
+	}
+	return op, nil
+}
 
-		switch {
-		case test.wantErr && err != nil:
-			continue
-		case test.wantErr && err == nil:
-			t.Errorf("Test %q - got nil, want error", test.description)
-		case !test.wantErr && err != nil:
-			t.Errorf("Test %q - got error, want nil: %v", test.description, err)
-		case !reflect.DeepEqual(got, test.want):
-			t.Errorf("Test %q - got\n%s\nwant\n%s", test.description, got, test.want)
-		}
+func runOpTestTransform(op transformOperation, test opTests) (interface{}, error) {
+	got, err := op.transform(test.in)
+
+	switch {
+	case test.wantErr && err != nil:
+		return got, err
+	case test.wantErr && err == nil:
+		return got, fmt.Errorf("Test %q - got nil, want error", test.description)
+	case !test.wantErr && err != nil:
```

Comment:
Neither this or the above case actually compares the errors, just that there WAS an error. It would be better if we compared the `err.Error()` values against each other to make sure we're getting the *right* error